### PR TITLE
Add support for right sidebar in public page

### PR DESCRIPTION
**Pull request by danxuliu**
This pull request adds the right sidebar also to the public page. Right now the right sidebar in the public page does not provide much value, but this is a preparatory step to later add a chat tab in the sidebar.

The guest name is now shown in the sidebar too instead of in the header as before. This made possible to move the _Enter/Exit fullscreen_ button to the top right corner (which in my opinion looks better), extend the sidebar to the top (as the header is hidden during calls in the public page), and reduce the UI elements shown during calls.

The room name and the guest name had a lot of similar code, so it was refactored into a new view, `EditableTextLabel`. An `EditableTextLabel` shows a single text attribute from a Backbone model and, optionally, also makes possible to edit it. As the guest name is stored in the local storage of the browser a new Backbone model that acts as a bridge between the local storage and Marionette views was introduced (`LocalStorageModel`).

### CODE REVIEW

File: js/views/callinfoview.js [hunk 1]
```diff
@@ -64,75 +60,128 @@
...
+
+			this._updateNameEditability();
+
+			this._guestNameEditableTextLabel = new OCA.SpreedMe.Views.EditableTextLabel({
+				model: this.getOption('guestNameModel'),
+				modelAttribute: 'nick',
+
+				extraClassNames: 'guest-name',
+				labelTagName: 'p',
+				labelPlaceholder: t('spreed', 'Guest'),
```
Comment (nickvergessen) [inline js/views/callinfoview.js]: Should we use something like "Set your name" here? Making it more pressuring to set a name?

**Comment by nickvergessen** [comment]
I have no idea about the JS splitting and organizing, but it works.
So would be fine by me 👍

File: js/views/callinfoview.js (hunk 1 above)
Comment (nickvergessen) [reply js/views/callinfoview.js]: Or can we add a label next to it?

Especially when the room has no name set, you see "Admin, You" and next line "Guest". This is a bit confusing.

Comment (danxuliu) [reply js/views/callinfoview.js]: Yes, I agree that just the label there with _Guest_ is terrible.

@jancborchardt Any advice on this? Should I add another label next to it like _Your name:_ or simply show _Set your name_ in the current label when there is no name set? Any other alternative?

Comment (jancborchardt) [reply js/views/callinfoview.js]: Yeah, the placeholder should be »Your name …«

Comment (danxuliu) [reply js/views/callinfoview.js]: Thanks, done!

**Comment by schiessle** [comment]
@nickvergessen @Ivansss can you review this pull request. I think we need it for chat. So it would be good to have this in to move on

**Pull request merged**